### ISSUE TITLE
Texture doesn't update in blender 3.1

### ISSUE THREAD

**Issue by lampysprites** [lampysprites/pribambase#23]
**Reality**
Image changes in blender are ignored until something else flushes them, e.g saving and reloading the image.

**Expectation**
textures change every time image data cahnges

**Logs**
In blender, please click `Window > Toggle System Console` and see if there's anything suspicious there. On Windows, copying from there is done by selecting text and right click.
```
To paste it here, use triple ticks like this
```

**Version info**
blender 3.1

**Additional context**
blender fuck up. seems that the new partial image update system is buggy, [e.g. like this](https://developer.blender.org/rBeec1c162000546d64232ad63e89cb0b54ac04881).

E.g. this test works in older versions, but not 3.1. Should fill image with random shade of gray on each call, but in 3.1 only works forst time; than need e.g. to save and reload the image.

```py
import bpy
import numpy as np
from random import random

try:
    img = bpy.data.images["Untitled"]
except KeyError:
    img = bpy.data.images.new("Untitled", 32, 32, alpha=True)


gen = np.ones((img.size[1], img.size[0], img.channels), dtype=np.float32) * random()
gen[:,:,3] = 1.0
img.pixels.foreach_set(np.ravel(gen))


img.update()
img.update_tag()
for area in bpy.context.screen.areas:
    area.tag_redraw()

```

**Comment by lampysprites** [comment]
save&reload workaround available in 2.2.2... now let's wait a few years as usual until blender fixes it on their side

**Comment by lampysprites** [comment]
Seem to work in 3.2 release, thankfully. Great. Awesome.
